### PR TITLE
[Feature] Allow actions to be shared via public link: Milestone 2

### PR DESCRIPTION
**Pull request by calherries**
This PR targets milestone 2 of "Allow actions to be shared via public link".

FE changes:
- https://github.com/metabase/metabase/pull/27675

For the BE: 
- Adds GET: /api/action/public: returns a list of shared actions similar to how we're currently returning shared questions, for the `/admin/settings/public-sharing` page

Epic: https://github.com/metabase/metabase/issues/27626

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27845)

### CODE REVIEW

File: frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx [hunk 1]
```diff
@@ -7,7 +7,6 @@ import {
   waitFor,
   waitForElementToBeRemoved,
   getIcon,
-  queryIcon,
```
Comment (calherries) [inline frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.unit.spec.tsx:10]: This was somehow unused and preventing me from committing, so I removed it.

**Review by qnkhuat — CHANGES_REQUESTED** [review]
I should have mentioned this in the milestone 1 PR, but application permissions should have tests in https://github.com/metabase/metabase/blob/master/enterprise/backend/test/metabase_enterprise/advanced_permissions/api/setting_test.clj

**Comment by calherries** [comment]
@qnkhuat Good catch, I'll add those today

**Review by qnkhuat — APPROVED** [review]

**Comment by calherries** [comment]
Reminder TODO: once Tim's work on https://github.com/metabase/metabase/pull/27866 is merged, I'll have to make sure the endpoint doesn't return archived actions

**Pull request merged**